backend/database/SQL script/init.sql

```diff
@@ -117,80 +124,83 @@ CREATE TABLE Weekly_Work_Schedules (
 );
 
-
-
 CREATE TABLE Intervals (
 	scheduleId 	INTEGER REFERENCES Schedules,
 	startTime	TIMESTAMP,
 	endTime	    TIMESTAMP,
     PRIMARY KEY (scheduleId)
 );
 
-CREATE TABLE Delivery_Details (
-	deliveryId			        INTEGER,
-	userId				        INTEGER NOT NULL, 
-	departTimeForRestaurant	    TIMESTAMP,
-	departTimeFromRestaurant    TIMESTAMP,
-	deliveryTimetoCustomer	    TIMESTAMP,
-	arrivalTimeAtRestaurant	    TIMESTAMP,  
-	ratingId			        INTEGER, 
-	ratingContent			    INTEGER,
-    PRIMARY KEY (deliveryId)	
-);
 
+--able to share the same promoCode
 CREATE TABLE Promotions (
 	promoCode	    VARCHAR(20),	
 	promoDesc 		VARCHAR(200),
-	createdBy	    VARCHAR(50),
-	applicableTo	VARCHAR(200),
-	discUnit	    VARCHAR(20),
-	discRate	    VARCHAR(20),
-	startDate	    DATE,
-	endDate	        DATE,
-	type		    VARCHAR(50),
+	createdBy	    VARCHAR(50), --?
+	applicableTo	VARCHAR(200) REFERENCES Restaurants(rname) ON DELETE CASCADE,
+	discUnit	    VARCHAR(20) NOT NULL,
+	discRate	    VARCHAR(20) NOT NULL,
+	startDate	    TIMESTAMP NOT NULL,
+	endDate	        TIMESTAMP NOT NULL,
 	PRIMARY KEY (promoCode, applicableTo)
 );
 
-CREATE TABLE Order_Details (
+CREATE TABLE Orders (
 	orderId 		INTEGER,
-	userId 			INTEGER NOT NULL REFERENCES Customers,
-	deliveryId		INTEGER NOT NULL REFERENCES Delivery_Details,
+	userId 			INTEGER NOT NULL REFERENCES Customers ON DELETE CASCADE ON UPDATE CASCADE,
 	promoCode		VARCHAR(20),
     applicableTo	VARCHAR(200),
-	rname			VARCHAR(100),
-	fname 			VARCHAR(100),
 	modeOfPayment 	VARCHAR(10) NOT NULL,
 	totalFoodPrice 	NUMERIC(8,2) NOT NULL,
 	deliveryFee		NUMERIC(8,2) NOT NULL,
 	timeOfOrder		TIMESTAMP NOT NULL,
-	foodQty		    INTEGER NOT NULL,
 	deliveryLocation	VARCHAR(100) NOT NULL,
-	usedRewardPoints	INTEGER NOT NULL,
+	usedRewardPoints	INTEGER DEFAULT 0,
 	givenRewardPoints	INTEGER NOT NULL,
-	rvId			INTEGER NOT NULL,
-	content		    VARCHAR(100),
+	reviewContent		    VARCHAR(100),
 	
-	PRIMARY KEY(orderId, rname, fname),
+	PRIMARY KEY(orderId),
 	FOREIGN KEY(promoCode, applicableTo)  REFERENCES Promotions,
-	FOREIGN KEY(rname, fname) REFERENCES Sells(rname, fname),
 	CHECK(modeOfPayment = 'cash' OR
 		 	modeOfPayment ='credit')
 );
 
+CREATE TABLE Order_Details (
+	orderId 		INTEGER REFERENCES Orders ON DELETE CASCADE ON UPDATE CASCADE,
+	rname			VARCHAR(100),
+	fname 			VARCHAR(100),
+	foodQty		    INTEGER NOT NULL,
+
+	PRIMARY KEY(orderId, rname, fname),
+	FOREIGN KEY(rname, fname) REFERENCES Sells(rname, fname),
+	CHECK(foodQty >= 1)
+);
+
+CREATE TABLE Delivery_Details (
+	orderId			        	INTEGER REFERENCES Orders ON DELETE CASCADE ON UPDATE CASCADE,
+	userId				        INTEGER NOT NULL, 
+	departTimeForRestaurant	    TIMESTAMP,
+	departTimeFromRestaurant    TIMESTAMP,
+	arrivalTimeAtRestaurant	    TIMESTAMP,
+	deliveryTimetoCustomer	    TIMESTAMP,
+	rating					    INTEGER,
+    PRIMARY KEY (orderId),
+	FOREIGN KEY (userId) REFERENCES Riders on DELETE CASCADE,
+	CHECK(rating <= 5)
+);
+
```

Comment:
ps do u mind explaining the diff between orders and order_details now? thanks !!

Reply:
In original schema, when you have multiple items in the same order. Order_details will have similar attributes for the same value. Lets say the order details have 1 order of different ood from same restaurant. 
```
INSERT INTO Order_details(orderId, fname, rname, quantity, otherattributes) VALUES
(1,'chicken', 'Res1', 1, XXX),
(1, 'duck', 'Res1', 2, XXX);
```
All XXX attributes that are common for different Order_details are repeated. Going by NF/BCNF. A->BCDE and A->BC, we can break it down into A->BC && A->DE. Which I'm doing now. 

```
INSERT INTO Orders(orderId, userId, promoCode, applicableTo, 
                modeOfPayment, totalFoodPrice, deliveryFee, timeOfOrder, deliveryLocation, 
                usedRewardPoints, givenRewardPoints, reviewContent) VALUES
(1, 1, NULL, NULL, 'cash', 30.00, 10.00, '2020-06-22 19:05:25', 'blk singapore', NULL, 3, 'SOO GOOD'),
(2, 1, NULL, NULL, 'cash', 40.00, 10.00, '2020-06-22 19:05:25', 'blk singapore', NULL, 3, 'NOT GOOD'),
(3, 2, NULL, NULL, 'credit', 20.00, 7.50, '2020-06-22 19:15:25', 'blk 123', NULL, 3, 'NOT GOOD1'),
(4, 2, NULL, NULL, 'credit', 20.00, 7.50, '2020-06-22 19:15:25', 'blk 123', NULL, 3, 'CHICKEN GOOD DUCK BAD1'),
(5, 3, NULL, NULL, 'credit', 20.00, 7.50, '2020-06-22 19:15:25', 'blk 123', NULL, 3, 'CHICKEN GOOD DUCK BAD2'),
(6, 3, NULL, NULL, 'credit', 20.00, 7.50, '2020-06-22 19:15:25', 'blk 123', NULL, 3, 'CHICKEN GOOD DUCK BAD3'),
(7, 3, NULL, NULL, 'credit', 20.00, 7.50, '2020-06-22 19:15:25', 'blk 123', NULL, 3, 'CHICKEN GOOD DUCK BAD4');
```

```
INSERT INTO Order_Details(orderId, rname, fname, foodQty) VALUES
(1, 'JIT YONG RESTAURANT', 'CHICKEN', 2),
(1, 'JIT YONG RESTAURANT', 'DUCK', 1),
(2, 'JIT YONG RESTAURANT1', 'CHICKEN', 3),
(3, 'JIT YONG RESTAURANT', 'DUCK', 1),
(4, 'JIT YONG RESTAURANT', 'DUCK', 1),
(5, 'JIT YONG RESTAURANT', 'DUCK', 1),
(6, 'JIT YONG RESTAURANT', 'DUCK', 1),
(7, 'JIT YONG RESTAURANT', 'DUCK', 1);
```

Reply:
oh i see!! its for the bcnf & nf i see.. thanks good work CP